src/metalperformanceshaders.cs

```diff
@@ -3403,12 +3449,249 @@ interface MPSNNMultiplicationNode {
+		float NeuronParameterB { get; }
+
+		[Export ("neuronParameterC")]
+		float NeuronParameterC { get; }
+
+		[Export ("setNeuronToPReLUWithParametersA:")]
+		void SetNeuronToPReLU (NSData parametersA);
+
+		[Export ("initWithDevice:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device);
+
+		// Keeping the same name as in the parent class so it ends up in an overload
+		[Export ("encodeToCommandBuffer:inputMatrix:biasVector:resultMatrix:")]
+		void EncodeToCommandBuffer (IMTLCommandBuffer commandBuffer, MPSMatrix inputMatrix, [NullAllowed] MPSVector biasVector, MPSMatrix resultMatrix);
+
+		[Export ("initWithCoder:device:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (NSCoder aDecoder, IMTLDevice device);
+
+		[Export ("copyWithZone:device:")]
+		[return: Release ()]
+		MPSMatrixNeuron CopyWithZone ([NullAllowed] NSZone zone, [NullAllowed] IMTLDevice device);
+	}
+
+	[TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (MPSMatrixSoftMax))]
+	[DisableDefaultCtor]
+	interface MPSMatrixLogSoftMax {
+
+		[Export ("initWithDevice:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device);
+
+		[Export ("initWithCoder:device:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (NSCoder aDecoder, IMTLDevice device);
+	}
+
+	[TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (MPSMatrixBinaryKernel))]
+	[DisableDefaultCtor]
+	interface MPSMatrixFullyConnected {
+
+		[Export ("sourceNumberOfFeatureVectors")]
+		nuint SourceNumberOfFeatureVectors { get; set; }
+
+		[Export ("sourceInputFeatureChannels")]
+		nuint SourceInputFeatureChannels { get; set; }
+
+		[Export ("sourceOutputFeatureChannels")]
+		nuint SourceOutputFeatureChannels { get; set; }
+
+		[Export ("alpha")]
+		double Alpha { get; set; }
+
+		[Export ("setNeuronType:parameterA:parameterB:parameterC:")]
+		void SetNeuronType (MPSCnnNeuronType neuronType, float parameterA, float parameterB, float parameterC);
+
+		[Export ("neuronType")]
+		MPSCnnNeuronType NeuronType { get; }
+
+		[Export ("neuronParameterA")]
+		float NeuronParameterA { get; }
+
+		[Export ("neuronParameterB")]
+		float NeuronParameterB { get; }
+
+		[Export ("neuronParameterC")]
+		float NeuronParameterC { get; }
+
+		[Export ("initWithDevice:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device);
+
+		// Keeping the same name as in the parent class so it ends up in an overload
+		[Export ("encodeToCommandBuffer:inputMatrix:weightMatrix:biasVector:resultMatrix:")]
+		void EncodeToCommandBuffer (IMTLCommandBuffer commandBuffer, MPSMatrix inputMatrix, MPSMatrix weightMatrix, [NullAllowed] MPSVector biasVector, MPSMatrix resultMatrix);
+
+		[Export ("initWithCoder:device:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (NSCoder aDecoder, IMTLDevice device);
+
+		[Export ("copyWithZone:device:")]
+		[return: Release ()]
+		MPSMatrixFullyConnected CopyWithZone ([NullAllowed] NSZone zone, [NullAllowed] IMTLDevice device);
```

Comment:
-> `Copy`
this is a specialized version of the NSCopying protocol required method